django/db/models/query.py

```diff
@@ -495,8 +494,7 @@ def _create_object_from_params(self, lookup, params):
         Used by get_or_create and update_or_create
         """
         try:
-            with transaction.atomic(using=self.db):
-                obj = self.create(**params)
+            obj = self.create(**params)
```

Comment:
Also, the atomic(savepoint=False) is really cheap (no DB action at all) when used inside another atomic. So, the atomic() here makes the atomic() inside save cheap enough to not matter (of course, benchmarks tell the truth).